docs/sources/logql/_index.md

```diff
@@ -277,6 +278,60 @@ will get those labels extracted:
 "status" => "200"
 ```
 
+##### pattern
+
+The pattern parser allow to explicitly extract fields from your log lines by defining a pattern expression which matches how the log line is structured.
+
+For example the following nginx log line:
+
+```log
+0.191.12.2 - - [10/Jun/2021:09:14:29 +0000] "GET /api/plugins/versioncheck HTTP/1.1" 200 2 "-" "Go-http-client/2.0" "13.76.247.102, 34.120.177.193" "TLSv1.2" "US" ""
+```
+
+Can be parsed with the expression:
+
+`<ip> - - <_> "<method> <uri> <_>" <status> <size> <_> "<agent>" <_>`
+
+and will extract the following fields
+
+```kv
+"ip" => "0.191.12.2"
+"method" => "GET"
+"uri" => "/api/plugins/versioncheck"
+"status" => "200"
+"size" => "2"
+"agent" => "Go-http-client/2.0"
+```
+
+A pattern expression is composed of captures and literals.
+
+A capture is noted `<example>` where `example` is the name of the field that will be extracted.
+Captures can be unnamed using `<_>`, in which case the capture skips matched content.
+
+Captures are matched from the beginning or the previous set of literals, to the end or the next set of literals.
+If a capture is not matched, the pattern parser will stop.
+
+Literals can be any sequence of UTF-8 characters, including whitespace.
+
+By default a pattern expression is anchored at the start, this means if the expression start with literals, the log line must also start with the same set of literals. You can use `<_>` at the beginning if you do not want to anchor your expression at the start.
+
+For example, given the following log line:
+
+```log
+level=debug ts=2021-06-10T09:24:13.472094048Z caller=logging.go:66 traceID=0568b66ad2d9294c msg="POST /loki/api/v1/push (204) 16.652862ms"
+```
+
+If you want to match from `msg="`, you can use the following expression:
+
+```pattern
+<_> msg="<method> <path> (<status>) <latency>"
+```
```

Comment:
```suggestion
Consider the log line

```log
level=debug ts=2021-06-10T09:24:13.472094048Z caller=logging.go:66 traceID=0568b66ad2d9294c msg="POST /loki/api/v1/push (204) 16.652862ms"
```

To match `msg="`, use the expression:

```pattern
<_> msg="<method> <path> (<status>) <latency>"
```
```

Reply:
hm the suggestion didn't work out